src/transformState.js

```diff
@@ -5,7 +5,38 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        Object.assign(state, extraData);
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
```

Comment:
We can make a separate function for this, so our `switch-case` will be smaller and more readable